### PR TITLE
Feature/add deployment tests

### PR DESCRIPTION
**Pull request by dudizimber**
## Summary by CodeRabbit

- **New Features**
	- Introduced new Kubernetes services for enhanced load balancing and port configurations.
	- Added support for standalone and replica database testing scenarios.
	- Implemented network policies for improved security and communication between pods.
	- Enhanced test environment setup with additional parameters for Redis and FalkorDB connections.

- **Bug Fixes**
	- Updated container image for FalkorDB backup to ensure compatibility.

- **Refactor**
	- Streamlined environment variable usage for database configurations.
	- Removed unused configurations related to `sentinel.service`.

- **Documentation**
	- Updated output declarations to reflect new and removed configurations, improving clarity on available resources and namespaces.

### CODE REVIEW

File: .github/workflows/gcp-full-infra-test-apply.yaml [hunk 1]
```diff
@@ -108,7 +108,7 @@ jobs:
           python3 -m venv .venv
           source .venv/bin/activate
           pip install -r requirements.txt
-          pytest
+          pytest -s tofu/gcp/test_env/__tests__/read-write/main.py --hostname ${{ env.FALKORDB_HOST }} --port ${{ env.FALKORDB_PORT }} --password ${{ secrets.TF_VAR_FALKORDB_PASSWORD }}
```
Comment (dudizimber) [inline .github/workflows/gcp-full-infra-test-apply.yaml:111]: Fix tests folder name

File: tofu/gcp/tenant/resources/k8s/falkordb-deployment/falkordb-deployment-multi-zone/main.tf [hunk 2]
```diff
@@ -75,6 +75,13 @@ resource "helm_release" "falkordb" {
     value = var.redis_port
   }
 
+  ###### MASTER ######
```
Comment (dudizimber) [inline tofu/gcp/tenant/resources/k8s/falkordb-deployment/falkordb-deployment-multi-zone/main.tf:78]: Add comment on why master is needed

File: tofu/gcp/tenant/resources/k8s/falkordb-deployment/falkordb-deployment-single-zone/main.tf [hunk 3]
```diff
@@ -89,6 +89,12 @@ resource "helm_release" "falkordb" {
     value = var.redis_port
   }
 
+  ###### MASTER ######
```
Comment (dudizimber) [inline tofu/gcp/tenant/resources/k8s/falkordb-deployment/falkordb-deployment-single-zone/main.tf:92]: Add comment on why master is needed

File: tofu/gcp/tenant/resources/k8s/policies/main.tf [hunk 4]
```diff
@@ -144,6 +146,116 @@ resource "kubernetes_network_policy" "allow_pod_backup" {
 }
 
 
+# Allow backup pod to connect to deployment
```
Comment (dudizimber) [inline tofu/gcp/tenant/resources/k8s/policies/main.tf:149]: Fix comment

File: tofu/gcp/test_env/__tests__/standalone/conftest.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    # This is called for every test. Only get/set command line arguments
+    # if the argument is specified in the list of test "fixturenames".
+    hostname = metafunc.config.option.hostname
+    if 'hostname' in metafunc.fixturenames and hostname is not None:
+        metafunc.parametrize("hostname", [hostname])
+    
+    sentinel_port = metafunc.config.option.sentinel_port
+    if 'sentinel_port' in metafunc.fixturenames and sentinel_port is not None:
+        metafunc.parametrize("sentinel_port", [sentinel_port])
+
```
Comment (dudizimber) [inline tofu/gcp/test_env/__tests__/standalone/conftest.py:18]: Not needed

File: tofu/gcp/test_env/__tests__/standalone/conftest.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+def pytest_addoption(parser):
+    parser.addoption("--hostname", default="localhost")
+    parser.addoption("--port", default="6379")
+    parser.addoption("--sentinel_port", default="26379")
```
Comment (dudizimber) [inline tofu/gcp/test_env/__tests__/standalone/conftest.py:4]: Not needed

File: tofu/gcp/test_env/main.tf [hunk 7]
```diff
@@ -102,6 +103,8 @@ module "standalone_tenant" {
     enable     = false,
     multi_zone = false
   }
+
+  labeler_image = "dudizimber/redis-pod-labeler:latest"
```
Comment (dudizimber) [inline tofu/gcp/test_env/main.tf:107]: Change repository name to falkordb

File: tofu/gcp/test_env/main.tf [hunk 8]
```diff
@@ -131,6 +135,8 @@ module "single_zone_tenant" {
     enable     = true,
     multi_zone = false
   }
+
+  labeler_image = "dudizimber/redis-pod-labeler:latest"
```
Comment (dudizimber) [inline tofu/gcp/test_env/main.tf:139]: Change repository name to falkordb

File: tofu/gcp/test_env/main.tf [hunk 9]
```diff
@@ -160,4 +167,5 @@ module "multi_zone_tenant" {
     enable     = true,
     multi_zone = true
   }
+  labeler_image = "dudizimber/redis-pod-labeler:latest"
```
Comment (dudizimber) [inline tofu/gcp/test_env/main.tf:170]: Change repository name to falkordb

**Review by AviAvni — DISMISSED** [review]

**Pull request merged**